create_geometry_3D.py

```diff
@@ -0,0 +1,258 @@
+    surfaces = gmsh.model.geo.addPlaneSurface([1], 1)
+    gmsh.model.setPhysicalName(2, surfaces, name_label)
+    return surfaces
+
+
+def create_volume(surface_id, depth):
+    """
+    creates volume by extruding 2D surface
+    :param surface_id: surface tag
+    :param depth: depth of 3D geometry
+    :return: -
+    """
+    gmsh.model.geo.extrude([(2, surface_id)], 0, 0, depth)
+
+
+def make_elements(point_coordinates, point_pairs, mesh_size, name_label):
+    """
+    takes point_pairs and puts their tags as the beginning and end of line in gmsh to create line
+    then creates surface
+    :param point_coordinates: point coordinates
+    :param point_pairs: has been generated by storing point tags of two consecutive points in an array
+    :param mesh_size: mesh size
+    :param name_label: surface name label from user input
+    :return: creates points, lines and surface elements
+    """
+    for point in point_coordinates:
+        coordinate = [point[0], point[1], point[2]]
+        create_point(coordinate, mesh_size)
+
+    line_lists = []
+
+    for i in range(len(point_pairs)):
+        line = [point_pairs[i][0], point_pairs[i][1]]
+        line_lists.append(i + 1)
+        create_line(line)
+
+    surfaces = create_surface(line_lists, name_label)
+    return surfaces
+
+
+def make_geometry_3D(point_coordinates, point_pairs, mesh_size, depth, name_label):
+    """
+    creates 3D geometries by extruding the 2D surface
+    :param point_coordinates: geometry points coordinates
+    :param point_pairs: points paired for lines
+    :param mesh_size: mesh size
+    :param depth: depth of 3D geometry
+    :param name_label: surface name label from user input
+    :return: -
+    """
+    surfaces = make_elements(point_coordinates, point_pairs, mesh_size, name_label)
+    create_volume(surfaces, depth)
+
+
+def make_geometry_2D(point_coordinates, point_pairs, mesh_size, name_label):
+    """
+    creates 2D geometries
+    :param point_coordinates: geometry points coordinates
+    :param point_pairs: points paired for lines
+    :param mesh_size: mesh size
+    :param name_label: surface name label from user input
+    :return: -
+    """
+    make_elements(point_coordinates, point_pairs, mesh_size, name_label)
+
+
+def extract_mesh_data(dims):
+    """
+    gets gmsh output data
+    :param dims: geometry dimension (2=2D or 3=3D)
+    :return: geometry and mesh data: node tags, node coordinates, element types, element tags 0D, 1D, 2D, 3D
+    """
+    node_tags, node_coords, node_params = gmsh.model.mesh.getNodes()  # nodes, elements
+    elem_types, elem_tags, elem_node_tags = gmsh.model.mesh.getElements()
+    print(elem_types)
+    num_elem = sum(len(i) for i in elem_tags)
+    print(" - Mesh has " + str(len(node_tags)) + " nodes and " + str(num_elem) +
+          " elements")
+
+    num_nodes = len(node_tags)
+    coord = np.reshape(node_coords, (num_nodes, 3))
+
+    if elem_types[0] == 1:
```

Comment:
Can you make a function instead, which retrieves the number of nodes from  elem type : e.g.:

n_nodes = get_num_nodes_from_elem_type(elem_types[0]),

where in get_num_nodes_from_elem_type, you put the required if statements e.g.:

`#2 noded line`
if elem_type == 1: return 2
`# 3 noded triangle`
if  elem_type == 2: return 3